### PR TITLE
fix(rich-text-editor): DP-116205 fix line break related issues

### PR DESCRIPTION
**Pull request by braddialpad**
# fix(rich-text-editor): DP-116205 fix line break related issues

## Obligatory GIF (super important!)

Obligatory GIF

## :hammer_and_wrench: Type Of Change

These types will increment the version number on release:

- [x] Fix

## :book: Jira Ticket

https://dialpad.atlassian.net/browse/DP-116205

## :book: Description

Changed to a different method to submit line breaks using shift + enter in rich-text-editor

## :bulb: Context

We've been having various issues with the way enter / new lines are handled within rich-text-editor and have to keep putting in workaround hacks to get it to work how we want. (see jira ticket for latest weird issue). HardBreak extension seems to cause various unwanted issues and after browsing around stack overflow it seems the best solution may be to remove it entirely.

I had to fix this already as part of my hackathon project, so I'm moving the code from my hackathon project to the actual component. Appears to work much more reliably as it is using built in tiptap/prosemirror commands. Could be some unknown regression here so I'm going to check with the product teams before merging to ubervoice, and also do more testing on my own.

## :pencil: Checklist

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

For all Vue changes:

- [x] I have made my changes in Vue 2 and Vue 3. Note: you may sync your changes from Vue 2 to Vue 3 (or vice versa) using the `./scripts/dialtone-vue-sync.sh` script. Read docs here: [Dialtone Vue Sync Script](../packages/dialtone-vue3/.github/CONTRIBUTING.md#dialtone-vue-sync-script)

## :crystal_ball: Next Steps

update in ubervoice, confirm the problems are fixed.

## :link: Sources

https://github.com/ueberdosis/tiptap/issues/2755

### CODE REVIEW

File: packages/dialtone-vue2/components/rich_text_editor/rich_text_editor.vue [hunk 1]
```diff
@@ -348,6 +348,31 @@ export default {
...
+              ]);
+            },
+            Enter: () => {
+              self.$emit('enter');
+              return true;
+            },
+          };
+        },
+      });
+      if (!this.allowLineBreaks) {
```
Comment (ninamarina) [inline packages/dialtone-vue2/components/rich_text_editor/rich_text_editor.vue]: Previously, if `allowLineBreaks` was `true`, `shift + enter` would still work to generate a soft break, now it doesn't.

**Comment by ninamarina** [comment]
When copying a text that has soft breaks created with shift + enter, the line breaks are removed:

paste

**Comment by braddialpad** [comment]
> When copying a text that has soft breaks created with shift + enter, the line breaks are removed:

Good find. I have re-included the "HardBreak" extension but am no longer triggering it when pressing enter so we should still be able to support pasted hard breaks.

File: packages/dialtone-vue2/components/rich_text_editor/rich_text_editor.vue (hunk 1 above)
Comment (braddialpad) [reply packages/dialtone-vue2/components/rich_text_editor/rich_text_editor.vue]: thanks 👍. fixed

**Comment by iropolo** [comment]
This could be an improvement of editor but does not fix the issue. 
I can still reproduce copy/paste issue of https://dialpad.atlassian.net/browse/DP-116205

**Comment by braddialpad** [comment]
> This could be an improvement of editor but does not fix the issue. I can still reproduce copy/paste issue of https://dialpad.atlassian.net/browse/DP-116205

Hmm yeah as soon as I added the hardbreak extension back in the problem came up again... Will look into more tomorrow.

**Comment by ninamarina** [comment]
Seems like the developers of the extension are fixing this: https://github.com/ueberdosis/tiptap/issues/5679#issuecomment-2506461301

**Comment by braddialpad** [comment]
> Seems like the developers of the extension are fixing this: [ueberdosis/tiptap#5679 (comment)](https://github.com/ueberdosis/tiptap/issues/5679#issuecomment-2506461301)

oh nice.. yeah I was trying to find a workaround but sounds like they are on top of it.

**Comment by braddialpad** [comment]
> Seems like the developers of the extension are fixing this: [ueberdosis/tiptap#5679 (comment)](https://github.com/ueberdosis/tiptap/issues/5679#issuecomment-2506461301)

I've implemented the fix they put up in this PR into our own component. This can be removed once the new link-extension release is out but I'm putting it in here right now just for the purposes of fixing this in a timely manner.

Should work correctly now, contains hardbreak extension and allows for pasting links with hardbreaks.

**Comment by iropolo** [comment]
console error on implementation

image

**Comment by braddialpad** [comment]
> console error on implementation

hmm looks like they have already merged the fix and are preparing a release. I'll just wait for the fix.

**Comment by braddialpad** [comment]
The fix hasn't been released with extension-link yet, however it was released in linkifyjs. Just adding the correct version of that dependency has fixed the issue

**Comment by braddialpad** [comment]
To be clear on the new behaviour:

when allowLineBreaks is true:

- Enter creates a paragraph
- Shift + Enter creates a line break
- Ctrl/Cmd + Enter creates a line break

when allowLineBreaks is false:

- Enter makes no change and fires an event
- Shift + Enter creates a paragraph
- Ctrl/Cmd + Enter creates a line break

**Comment by braddialpad** [comment]
Ready for review.

**Comment by juliodialpad** [comment]
Still broken...

Now the link with soft breaks gets split

image

File: package.json [hunk 2]
```diff
@@ -124,6 +124,7 @@
     "date-fns": "2.30.0",
     "docopt": "0.6.2",
     "emoji-toolkit": "9.0.1",
+    "linkifyjs": "4.2.0",
```
Comment (juliodialpad) [inline package.json:127]: This is a temp solution while they release the proper fix, right?

**Review by juliodialpad — APPROVED** [review]
Looking good!

File: package.json (hunk 2 above)
Comment (braddialpad) [reply package.json:127]: yeah basically, they just need to update linkifyjs in extension-link and release it, then we can remove this.

**Comment by bianca-artola-dialpad** [comment]
hey @braddialpad i'm having some issues when using shift + enter on the message input on beta

https://github.com/user-attachments/assets/519c4b7d-1f5c-45f7-aaa0-6cca0015f707

```
Error: There is no node type named 'listItem'. Maybe you forgot to add the extension?
```

is there something we need to add on firespotter side to make it work?

**Comment by braddialpad** [comment]
> hey @braddialpad i'm having some issues when using shift + enter on the message input on beta
> 
>  Screen.Recording.2024-12-17.at.12.41.32.PM.mov 
> ```
> Error: There is no node type named 'listItem'. Maybe you forgot to add the extension?
> ```
> 
> is there something we need to add on firespotter side to make it work?

@bianca-artola-dialpad strangely I'm not getting this at all when I try. Are you still getting it on beta?

**Comment by bianca-artola-dialpad** [comment]
Not reproducible now on beta but I think is because of the [new release process](https://dialpad.atlassian.net/wiki/x/QoFMIw) for beta in which we rollbacked to the previous release on Tuesday (so no front-end changes will be displayed for some hours). I'm having this issue when running it locally, but we can wait some hours after beta has the latest front-end changes

**Comment by braddialpad** [comment]
> Not reproducible now on beta but I think is because of the [new release process](https://dialpad.atlassian.net/wiki/x/QoFMIw) for beta in which we rollbacked to the previous release on Tuesday (so no front-end changes will be displayed for some hours). I'm having this issue when running it locally, but we can wait some hours after beta has the latest front-end changes

Ah yeah that must be it.. I'll try locally.

**Comment by braddialpad** [comment]
Found the problem, fix coming shortly.

**Pull request merged**